src/QsCompiler/Compiler/CompilationLoader.cs

```diff
@@ -549,7 +554,7 @@ public CompilationLoader(SourceLoader loadSources, ReferenceLoader loadReference
             buildProperties.Add(MSBuildProperties.ResolvedProcessorArchitecture, processorArchitecture);
 
             var compilationManager = new CompilationUnitManager(
-                new ProjectProperties(buildProperties),
+                new ProjectProperties(buildProperties, this.config.WarningsAsErrors),

```

Comment:
There are two scenarios you have to handle: when the compiler is used as a command line program and when it's running as a language server. This handles the CLI case, but I don't think WarningsAsErrors will be initialized properly in the language server because the extra argument is omitted here:

https://github.com/microsoft/qsharp-compiler/blob/2227253bf5b0791d616d3e6b60f592b5db062bc0/src/QsCompiler/CompilationManager/ProjectManager.cs#L159

Reply:
OK. I accept it in build properties. Command line options are also written to the build properties.